Modules/_ctypes/callproc.c

```diff
@@ -1019,7 +1019,7 @@ void _ctypes_extend_error(PyObject *exc_class, const char *fmt, ...)
 
     PyErr_Fetch(&tp, &v, &tb);
     PyErr_NormalizeException(&tp, &v, &tb);
-    cls_str = PyObject_Str(tp);
+    cls_str = PyUnicode_FromString(((PyTypeObject *)tp)->tp_name);
```

Comment:
You can use `PyType_GetName` here.

Reply:
That is much better! Thank you, I really like this suggestion 🙌